libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java

```diff
@@ -322,29 +321,31 @@ private String getSecretValue(String secretId) {
         try {
             return secretsProvider.get(secretId);
         } catch (DecryptionFailureException e) {
-            LogHelper.logErrorMessage(
-                    "Secrets manager failed to decrypt the protected secret using the configured KMS key",
-                    e.getMessage());
+            LOGGER.error(
+                    LogHelper.buildErrorMessage(
+                            "Secrets manager failed to decrypt the protected secret using the configured KMS key",
+                            e));
         } catch (InternalServiceErrorException e) {
-            LogHelper.logErrorMessage(
-                    "Internal server error occurred with Secrets manager", e.getMessage());
+            LOGGER.error(
+                    LogHelper.buildErrorMessage(
+                            "Internal server error occurred with Secrets manager", e));
         } catch (InvalidParameterException e) {
             LOGGER.error(
-                    "An invalid value was provided for the param value: {}, details: {}",
-                    secretId,
-                    e.getMessage());
+                    LogHelper.buildErrorMessage(
+                            String.format(
+                                    "An invalid value was provided for the param value: %s",
+                                    secretId),
+                            e));
         } catch (InvalidRequestException e) {
-            LogHelper.logErrorMessage(
-                    "Parameter value is not valid for the current state of the resource",
-                    e.getMessage());
+            LOGGER.error(
+                    LogHelper.buildErrorMessage(
+                            "Parameter value is not valid for the current state of the resource",
+                            e));
         } catch (ResourceNotFoundException e) {
             LOGGER.error(
-                    new StringMapMessage()
-                            .with(
-                                    LOG_MESSAGE_DESCRIPTION.getFieldName(),
-                                    "Failed to find the resource within Secrets manager.")
-                            .with(LOG_SECRET_ID.getFieldName(), secretId)
-                            .with(LOG_ERROR_DESCRIPTION.getFieldName(), e.getMessage()));
+                    LogHelper.buildErrorMessage(
+                                    "Failed to find the resource within Secrets manager.", e)
+                            .with(LOG_SECRET_ID.getFieldName(), secretId));
```

Comment:
This is the other advantage of this pattern - we can easily extend the message with additional properties.

(There are almost certainly other examples where we can do this, but I haven't been through them...)

Reply:
Yeah this way we are adding the various attributes to the message field in the log message.
"message": {
    "description": "ContraIndicators Verifiable Credential validated.",
    "criId": "3123213"
  },